client/src/pages/TeacherPage.jsx

```diff
@@ -47,38 +73,73 @@ function TeacherPage(props)
     setOpenDialogApplication(true);
   };  
 
+  const [listTitles,setListTitles]=useState('');
+
+
   const fetchData = async () => {
-    if (user) {
+
+    if (user) 
+    {
+      
+       const updateExpiredStatus = (proposals) => {
+         proposals?.forEach(item => {
+           if (dayjs(item.expiration_date).isBefore(currentDataAndTime.subtract(1, 'day'))) {
```

Comment:
I think this comparison isn't that clear. Because you're comparing the proposal expiration_date which is only a date (YYYY-MM-dd) with date and time. So for example imagine today is 2023-12-10  at 14:00 and expiration_date is 2023-12-09 (I think that it will be converted to 00:00) then expiration_date is not before currentDate, which I think it was supposed to be, right? Since it is in the past. I would declare explicitly the time in expiration date or change a bit this comparison.